### PR TITLE
update templates for new scope attribute (initial commit)

### PR DESCRIPTION
**Pull request by jennyf19**
@jmprieur  let's discuss, i have a few questions.

### CODE REVIEW

File: ProjectTemplates/templates/ComponentsWebAssembly-CSharp/Server/Controllers/CallWebApiController.cs [hunk 1]
```diff
@@ -11,11 +11,9 @@ namespace AspNetCoreMicrosoftIdentityWebProjectTemplates.templates.ComponentsWeb
     [Authorize]
     [ApiController]
     [Route("[controller]")]
+    [RequiredScope("access_as_user", IsReusable = true)] // The web API will only accept tokens 1) for users, and 2) having the "access_as_user" scope for this API
```
Comment (jmprieur) [inline ProjectTemplates/templates/ComponentsWebAssembly-CSharp/Server/Controllers/CallWebApiController.cs:14]: As proposed in the other PR, let's embed IsReusable=true in the RequiredScope attribute itself (as this is a perf optimization)

We need to decide if we prefer:
```CSharp
[RequiredScope("access_as_user"] // The web API will only accept tokens 1) for users, and 2) having the "access_as_user" scope for this API
```

or 
```CSharp
[RequiredScope(CallWebApiController.scopeRequiredByApi]
public class CallWebApiController : Controller
    {
        // The web API will only accept tokens 1) for users, and 2) having the "access_as_user" scope for this API
        public const string[] scopeRequiredByApi = new string[] { "access_as_user" };
```

File: ProjectTemplates/templates/Worker-CSharp/GreeterService.cs [hunk 2]
```diff
@@ -40,10 +34,10 @@ public class GreeterService : Greeter.GreeterBase
         }
 
         [Authorize]
+        [RequiredScope("access_as_user", IsReusable = true)] // The web API will only accept tokens 1) for users, and 2) having the "access_as_user" scope for this API
```
Comment (jmprieur) [inline ProjectTemplates/templates/Worker-CSharp/GreeterService.cs:37]: Would it make sense to have it on the GreeterService?

**Review by jmprieur — COMMENTED** [review]
LGTM.
We have a couple of decisions to make.

File: ProjectTemplates/templates/ComponentsWebAssembly-CSharp/Server/Controllers/CallWebApiController.cs (hunk 1 above)
Comment (jennyf19) [reply ProjectTemplates/templates/ComponentsWebAssembly-CSharp/Server/Controllers/CallWebApiController.cs:14]: fixed in other pr, so will update

---
In reply to: [573604272](https://github.com/AzureAD/microsoft-identity-web/pull/946#discussion_r573604272) [](ancestors = 573604272)

**Pull request merged**